src/protocol.js

```diff
@@ -1429,7 +680,7 @@ function read(buffer, cursor, fieldInfo, rootNodes) {
     };
   }
   var readResults = type[0](buffer, cursor, fieldInfo.typeArgs, rootNodes);
-  if (readResults == null) {
+  if (readResults == null && fieldInfo.type!=="condition") {
```

Comment:
Then, you can remove this check here. A reader is never supposed to return null.